app/models/news_vote.rb

```diff
@@ -0,0 +1,104 @@
+class NewsVote < ApplicationRecord
+  upsert_keys [:news_item_id, :user_id]
+  belongs_to :news_item
+  belongs_to :user
+
+  enum vote: { neutral: 0, up: 1, down: -1 }
+  validates :vote, inclusion: { in: votes.keys }
+
+  def self.votes_for_news_item(default = nil, news_item: nil, news_item_id: nil)
+    news_item ||= default
+    if news_item.blank? && news_item_id.blank?
+      return
+    end
+
+    @news_item_id = news_item.try(:id) || news_item_id
+    votes = self.votes_by_news_item(news_item_ids: [@news_item_id])
+    votes.dig(@news_item_id) || self.vote_summary(id: @news_item_id)
+  end
+
+  def self.votes_by_news_item(default = nil, news_items: nil, news_item_ids: nil)
+    news_items ||= default
+    if news_items.blank? && news_item_ids.blank?
+      return
+    end
+
+    @news_item_ids = news_items.try(:map) { |item| item.try(:id) } || news_item_ids
+    # create default hash to ensure all ids are present even if no sql result is present
+    default_hash = @news_item_ids.reduce({}) { |hash, id| hash.merge(Hash[id, self.vote_summary(id: id)]) }
+
+    NewsVote.select(<<-SQL
+        news_item_id,
+        coalesce(sum(vote), 0) as total
+      SQL
+      )
+      .where('vote != 0')
+      .group(:news_item_id)
+      .having(news_item_id: @news_item_ids)
+      .reduce(default_hash) do |hash, item|
+        hash[item.news_item_id] = self.vote_summary(
+          id: item.news_item_id,
+          total: item.total,
+        )
+        hash
+      end
+  end
+
+  def self.vote_summary(id: nil, total: 0)
```

Comment:
Do we want to allow `nil` id?

Reply:
I presume the `id` refers to a `news_item_id`?

If so, would it be better to have it as an instance method on NewsItem instead?

If not, is the `id` for a `NewsVote`, and if that's the case, would it be better to have it as an instance method on `NewsVote` instead?